backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java

```diff
@@ -187,56 +206,129 @@ public RepositoryEntity getKey() {
+            return newLineageEntryBatchResponseSingleList.stream();
+          }).collect(Collectors.toList());
+      if (result.size() != 0) {
+        newList.add(LineageEntryBatchResponse.newBuilder().addAllItems(result).build());
+      }
+      return newList.stream();
+    }).collect(Collectors.toList());
+  }
+
+  private void checkResourcesExistsAndAccessible(Session session, List<LineageEntry> lineageEntries)
       throws ModelDBException, NoSuchAlgorithmException, InvalidProtocolBufferException {
     Set<String> experimentRuns = new HashSet<>();
     Map<Long, RepositoryContainer> blobs = new HashMap<>();
     for (LineageEntry lineageEntry : lineageEntries) {
-      switch (lineageEntry.getDescriptionCase()) {
-        case EXPERIMENT_RUN:
-          String experimentRun = lineageEntry.getExperimentRun();
-          if (!experimentRuns.contains(experimentRun)) {
-            boolean result = experimentDAO.isExperimentRunExists(session, experimentRun);
-            if (!result) {
-              throw new ModelDBException("Experiment run doesn't exist");
-            }
-            experimentRuns.add(experimentRun);
-          }
-          break;
-        case BLOB:
-          VersioningLineageEntry blob = lineageEntry.getBlob();
-          long repositoryId = blob.getRepositoryId();
-          RepositoryEntity repo;
-          CommitMap result;
-          if (!blobs.containsKey(repositoryId)) {
-            repo =
-                repositoryDAO.getRepositoryById(
-                    session, RepositoryIdentification.newBuilder().setRepoId(repositoryId).build());
-            blobs.put(repositoryId, new RepositoryContainer(repo));
-            result = blobs.get(repositoryId).getValue();
-          } else {
-            RepositoryContainer entityMapEntry = blobs.get(repositoryId);
-            repo = entityMapEntry.getKey();
-            result = entityMapEntry.getValue();
-          }
-          String commitSha = blob.getCommitSha();
-          Set<String> blobResult;
-          if (!result.containsKey(commitSha)) {
-            commitDAO.getCommitEntity(session, commitSha, session1 -> repo);
-            result.put(commitSha, new HashSet<>());
-          }
-          blobResult = result.get(commitSha);
-          String stringFromProtoObject =
-              ModelDBUtils.getStringFromProtoObject(
-                  Location.newBuilder().addAllLocation(blob.getLocationList()));
-          if (!blobResult.contains(stringFromProtoObject)) {
-            blobDAO.getCommitComponent(session1 -> repo, commitSha, blob.getLocationList());
-            blobResult.add(stringFromProtoObject);
-          }
-          break;
-        default:
-          throw new ModelDBException("Unexpected type", Code.INTERNAL);
-      }
+      validate(session, experimentRuns, blobs, lineageEntry);
+    }
+  }
+
+  private void validate(Session session, Set<String> experimentRuns,
+      Map<Long, RepositoryContainer> blobs, LineageEntry lineageEntry)
```

Comment:
`Map<Long, RepositoryContainer> blobs` is not blobs, but repositories, right?

Reply:
yes, fixed